src/main/java/org/terasology/anotherWorld/decorator/structure/AbstractMultiChunkStructureDefinition.java

```diff
@@ -22,16 +23,16 @@ protected AbstractMultiChunkStructureDefinition(PDist frequency) {
     }
 
     @Override
-    public final Collection<Structure> generateStructures(Vector3i chunkSize, long seed, BlockRegion region) {
+    public final Collection<Structure> generateStructures(Vector3ic chunkSize, long seed, BlockRegionc region) {
         List<Structure> result = new LinkedList<>();
         float maxRange = getMaxRange();
 
-        int chunksRangeToEvaluateX = (int) Math.ceil(maxRange / chunkSize.x);
-        int chunksRangeToEvaluateY = (int) Math.ceil(maxRange / chunkSize.y);
-        int chunksRangeToEvaluateZ = (int) Math.ceil(maxRange / chunkSize.z);
+        int chunksRangeToEvaluateX = (int) Math.ceil(maxRange / chunkSize.x());
+        int chunksRangeToEvaluateY = (int) Math.ceil(maxRange / chunkSize.y());
+        int chunksRangeToEvaluateZ = (int) Math.ceil(maxRange / chunkSize.z());
 
-        Vector3i minChunk = ChunkMath.calcChunkPos(region.getMin(new Vector3i()), new Vector3i());
-        Vector3i maxChunk = ChunkMath.calcChunkPos(region.getMax(new Vector3i()), new Vector3i());
+        Vector3i minChunk = Chunks.toChunkPos(region.getMin(new Vector3i()), new Vector3i());
```

Comment:
I think this is code smell, as we create new vectors although we don't need them... =/ In-place computation variants in `Chunks` would help - I think we need them... 
Otherwise, the solution would be
```suggestion
 		 Vector3i minRegion = region.getMin(new Vector3i();
        Vector3i minChunk = Chunks.toChunkPos(minRegion, minRegion);
```
which is quite bulky just to implement: "please compute the chunk position from the min vector of this region into a new vector"

Reply:
tl;dr: I think we should add in-place computations for `Chunks#...`

Reply:
...